File: eduBoostBackend/src/main/java/com/fptu/eduBoostBackend/service/FeedbackService.java
package com.fptu.eduBoostBackend.service;

import com.fptu.eduBoostBackend.dto.request.CreateFeedbackRequest;
import com.fptu.eduBoostBackend.dto.request.RespondFeedbackRequest;
import com.fptu.eduBoostBackend.dto.response.FeedbackResponse;
import com.fptu.eduBoostBackend.dto.response.FeedbackStatsResponse;

import java.util.List;

public interface FeedbackService {

    /** Teacher — submit new feedback */
    FeedbackResponse createFeedback(CreateFeedbackRequest request);

    /** Teacher — list my feedbacks */
    List<FeedbackResponse> getMyFeedbacks();

    /** Teacher — get single feedback */
    FeedbackResponse getMyFeedbackById(Long id);

    /** Admin — list all feedbacks */
    List<FeedbackResponse> getAllFeedbacks();

    /** Admin — feedback statistics */
    FeedbackStatsResponse getStats();

    /** Admin — get feedback detail */
    FeedbackResponse getFeedbackById(Long id);

    /** Admin — respond to feedback */
    FeedbackResponse respondToFeedback(Long id, RespondFeedbackRequest request);
}

File: eduBoostFrontend/src/services/feedbackService.js
import { apiClient } from './api';

const BASE = '/feedback';

export const feedbackService = {
    /* ─── Teacher ─── */
    create: (data) =>
        apiClient.post(BASE, data).then(r => r.data?.data ?? r.data),

    getMyFeedbacks: () =>
        apiClient.get(`${BASE}/my`).then(r => r.data?.data ?? r.data),

    getMyFeedbackById: (id) =>
        apiClient.get(`${BASE}/my/${id}`).then(r => r.data?.data ?? r.data),

    /* ─── Admin ─── */
    getAll: () =>
        apiClient.get(`${BASE}/admin/all`).then(r => r.data?.data ?? r.data),

    getStats: () =>
        apiClient.get(`${BASE}/admin/stats`).then(r => r.data?.data ?? r.data),

    getById: (id) =>
        apiClient.get(`${BASE}/admin/${id}`).then(r => r.data?.data ?? r.data),

    respond: (id, data) =>
        apiClient.put(`${BASE}/admin/${id}/respond`, data).then(r => r.data?.data ?? r.data),
};
